cypress/support/script_loader.js

```diff
@@ -87,7 +97,18 @@ function loadScriptsBeforeForUnitTests(...scriptKeys) {
     // are true, which will keep Cypress from proceeding until all that work is
     // done.
     for (const callback of callbacks) {
-      waitForCallback(callback);
+      const cypressPromise = waitForCallback(callback);
+      // If this script had dependent scripts, load them when this script has
+      // been loaded, and add the callback for those scripts in.
+      const scriptsAndCallbacks = extraScripts.get(callback);
+      if (scriptsAndCallbacks) {
+        cypressPromise.then(() => {
+          for (const script of scriptsAndCallbacks.scripts) {
+            addScriptToDocument(script);
+          }
+          return waitForCallback(scriptsAndCallbacks.callback);
```

Comment:
why return here?

Reply:
As we've discussed, I like returning promises/chainables when available. If someone adds another then clause after this, they won't have to change this line to return. Without the return, it would be a very subtle gotcha, because the next then wouldn't wait for the first then to complete. Avoiding that bug seems worth returning a useful value when available.